# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/composer/nsEditingSession.cpp
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a editor/composer/nsEditingSession.cpp
--- a/editor/composer/nsEditingSession.cpp	Fri Jul 31 13:37:00 2020 +0000
+++ b/editor/composer/nsEditingSession.cpp	Fri Jul 31 13:37:13 2020 +0000
@@ -184,7 +184,8 @@
   // Disable plugins in this document:
   mPluginsEnabled = aDocShell.PluginsAllowedInCurrentDoc();
 
-  aDocShell.GetBrowsingContext()->SetAllowPlugins(false);
+  rv = aDocShell.GetBrowsingContext()->SetAllowPlugins(false);
+  NS_ENSURE_SUCCESS(rv, rv);
 
   mDisabledJSAndPlugins = true;
 
@@ -211,9 +212,8 @@
   // Disable plugins in this document:
   auto* browsingContext = aWindow->GetBrowsingContext();
   NS_ENSURE_TRUE(browsingContext, NS_ERROR_FAILURE);
-  browsingContext->SetAllowPlugins(mPluginsEnabled);
 
-  return NS_OK;
+  return browsingContext->SetAllowPlugins(mPluginsEnabled);
 }
 
 /*---------------------------------------------------------------------------